plot/Makefile

```diff
@@ -0,0 +1,132 @@
+
+plotvalidatecer: $(TMP_FAST_LOG) $(TMP_100_ITERATIONS) $(TMP_CHECKPOINT) $(TMP_EVAL) $(TMP_VALIDATE) $(TSV_VALIDATE_CER)
+
+# Make TSV with CER at every 100 iterations.
+$(TMP_100_ITERATIONS): $(MODEL_LOG)
+	@echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@"
+	@grep 'At iteration' $(MODEL_LOG) \
+		| sed -e '/^Sub/d' \
+		| sed -e '/^Update/d' \
+		| sed -e '/^ New worst char/d' \
+		| sed -e 's/At iteration \([0-9]*\)\/\([0-9]*\)\/.*char train=/\t\t\1\t\2\t\t/' \
+		| sed -e 's/%, word.*/\t/' >>  "$@"
+
+# Make TSV with Checkpoint CER.
+$(TMP_CHECKPOINT): $(MODEL_LOG)
+	@echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@"
+	@grep 'best model' $(MODEL_LOG) \
+		| sed -e 's/^.*\///' \
+		| sed -e 's/\.checkpoint.*$$/\t\t\t/' \
+		| sed -e 's/_/\t/g' >>  "$@"
+
+# Make TSV with Eval CER.
+$(TMP_EVAL): $(MODEL_LOG)
+	@echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@"
+	@grep 'Eval Char' $(MODEL_LOG) \
+		| sed -e 's/^.*[0-9]At iteration //' \
+		| sed -e 's/,.* Eval Char error rate=/\t\t/'  \
+		| sed -e 's/, Word.*$$/\t\t/' \
+		| sed -e 's/^/\t\t/' >>  "$@"
+
+# Validation log is generated only when CER falls below $(VALIDATE_CER)].9, so check for file existence.
+# Extract Filename and CER from Validate log for building Validate TSV file.
+# Make TSV with Validate CER.
+$(TMP_VALIDATE): $(TMP_FAST_LOG)
+	@if [ ! -s $(TMP_FAST_LOG) ]; \
+	then \
+		echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@" ;\
+	else \
+		grep -E "$(VALIDATE_LIST).log$$|iteration" $(TMP_FAST_LOG) > $(TMP_VALIDATE_LOG) ;\
+		echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@" ;\
+		sed 'N;s/\nAt iteration 0, stage 0, /At iteration 0, stage 0, /;P;D'  $(TMP_VALIDATE_LOG) \
+			| grep 'Eval Char' \
+			| sed -e "s/.$(VALIDATE_LIST).log.*Eval Char error rate=/\t\t\t/" \
+			| sed -e 's/, Word.*$$//' \
+			| sed -e 's/\(^.*\)_\([0-9].*\)_\([0-9].*\)_\([0-9].*\)\t/\1\t\2\t\3\t\4\t/g' >> "$@" ;\
+	fi;
+
+# Build fast traineddata file list with CER in range [0-VALIDATE_CER].[0-9].
+FAST_DATA_FILES := $(shell find ../data/$(MODEL_NAME)/tessdata_fast/ -type f -name $(MODEL_NAME)_[0-$(VALIDATE_CER)].[0-9]*_*.traineddata | sort -n -r)
+
+# Build validate log files list based on above traineddata list.
+FAST_LOG_FILES := $(subst tessdata_fast,tessdata_fast,$(patsubst %.traineddata,%.$(VALIDATE_LIST).log,$(FAST_DATA_FILES)))
+
+#Note: This does not find the new traineddata files from current run. 
+#	Hence `make` needs to be run twice to generate new validate.log files.
+
+$(FAST_LOG_FILES): %.$(VALIDATE_LIST).log: %.traineddata
+	OMP_THREAD_LIMIT=1 time -p lstmeval  \
+		--verbosity=0 \
+		--model $< \
+		--eval_listfile $(TMP_VALIDATE_LIST) 2>&1 | grep "^At iteration" > $@
```

Comment:
This rule needs still to advertise all its dependencies, esp. `TMP_VALIDATE_LIST` and `FAST_DATA_FILES`.